src/sentry/taskworker/retry.py

```diff
@@ -0,0 +1,79 @@
+from __future__ import annotations
+
+from collections.abc import Sequence
+from multiprocessing.context import TimeoutError
+
+from sentry_protos.sentry.v1.taskworker_pb2 import RetryState
+
+
+class RetryError(Exception):
+    """
+    Exception that tasks can raise to indicate that the current task
+    should be retried.
+    """
+
+
+class Retry:
+    """Used with tasks to define the retry policy for a task"""
+
+    __times: int
+    __allowed_exception_types: Sequence[type] | None
+    __denied_exception_types: Sequence[type] | None
+    __deadletter: bool | None
+    __discard: bool | None
+
+    def __init__(
+        self,
+        times: int = 1,
+        on: Sequence[type] | None = None,
+        ignore: Sequence[type] | None = None,
+        deadletter: bool | None = None,
+        discard: bool | None = None,
```

Comment:
it may be beneficial to make these named-only arguments, especially because the latter ones are currently [boolean traps](https://youtu.be/CnRkXO_a5mI) and actually wait why are these _nullable_ boolean? that doesn't seem right